blog/2023-07-23-openhab-4-0-release.md

```diff
@@ -0,0 +1,251 @@
+
+The upcoming openHAB 4.1 release will probably improve the UI, so you can simply select which transformation you want to apply on an Item state, but until then the transformation editor provides guidance what to fill into the state description field.
+
+### Persistence Configuration
+
+Seems like openHAB 4 is full of new UI pages — It finally supports configuring persistence from the UI.
+
+As most UI configuration pages, the persistence configuration page is divided into two tabs: A design and a code tab.
+
+The design tab allows to graphically configure strategies and filters for selected Items, create new cron strategies (common ones are automatically created for you) and manage filters.
+openHAB 4 also has a few new filters you should check out!
+
+The code tab represents the persistence configuration in YAML code and is perfect for sharing, e.g. when asking for help on the forum, but everything can be configured without touching a single line of code.
+
+<!-- Add image here -->
+
+To try it yourself, open a persistence add-on’s settings from the add-on store and click on the blue “Persistence configuration” header.
+
+### Developer Sidebar Search
+
+In openHAB 3, the developer sidebar search was only able to find Items by name or label, Things by UID or label, rules by UID or name and pages by UID.
+
+With openHAB 4, we gave it a big upgrade:
+It is now possible to search inside nearly every UI entity. 
+No matter where, you find what you search for.
+
+Imagine you want to know which of your rules and pages use a specific Item. Just type the Item name in the developer sidebar search, and you’ll find all occurrences of the search query in pages, scenes, scripts and rules. 
+Even Blockly and the code of script actions are now searchable.
+
+<!-- Add image here -->
+
+For an exact list of entities the search checks your query against, please have a look at [the documentation](/docs/tutorial/tips-and-tricks.html#search).
+
+### New Input Field for Sitemap UIs
+
+_Laurent Garnier, openHAB Maintainer_
+
+No new user interface elements have been added for sitemap UIs since openHAB version 1.
+This finally changes with openHAB 4 with the introduction of a new UI element long-awaited by users: The `input` field.
+
+```java
+Input item=<itemname> [label="<labelname>"] [icon="<iconname>"] [inputHint="<inputHint>"]
+```
+
+This element allows entering of text, numbers and dates/times and updating the underlying items.
+Text, number or number with unit values and dates/times can be updated in the respective item types.
+
+The parameter `inputHint` will give a hint to the user interface to use a specific widget adapted to a specific use.
+For example, it can be a time picker or a date picker.  Valid values for the parameter are `text`, `number`, `date`, `time` and `datetime`.
+Support of these widget types vary by user interface and/or browser.
+
+This new input field is already supported in Basic UI and in the Android app.
```

Comment:
Since this is a blog, perhaps we could bring it to life with an example usage and an image? It could be your own example with a manual meter reading, or perhaps an input text field for sending a toast to your LGTV with the text: "Dinner is ready :-)" for some fun. Time is short, but if you like the idea, I might be able to do a quick sitemap for that and screenshot it from my Android phone.

Reply:
Please do it if you have time.

Reply:
Just for inspiration...

items/tv.items
```java
Switch LGTV_Power "Power" <switch> { channel="lgwebos:WebOSTV:tv:power" }
Player LGTV_Control "Control" { channel="lgwebos:WebOSTV:tv:mediaPlayer" }
String LGTV_Toast "Toast { channel="lgwebos:WebOSTV:tv:toast" }
```

sitemaps/tv.sitemap
```perl
sitemap multimedia label="TV" {
    Switch item=LGTV_Power
    Default item=LGTV_Control visibility=[LGTV_Power==ON]
    Input item=LGTV_Toast icon="screen" inputHint="text" visibility=[LGTV_Power==ON]
}
```

Android app:
![image](https://github.com/openhab/website/assets/19519842/4317d85e-0c72-446e-b2ed-2347feb0f394)

LG TV:
![image](https://github.com/openhab/website/assets/19519842/e16d9002-33e1-45bd-8705-99cd3fe0080b)

Reply:
Yes that is a fun example.

Reply:
It sure beats the automated toasts saying "Laundry is ready". 😉

Reply:
An example showing date picker would have been cool too.

Reply:
Do you mind creating a PR to include that example? 
I can add the missing UI screenshots meanwhile ...

Reply:
I have some issues trying to fork your repository, and also don't have permissions to push directly. Can you copy from there? Just replace --- by ```

```
There are many use cases for this feature.
For example, it can be used for entering manual meter readings - or perhaps drawing attention from your family:

items/tv.items
---java
Switch LGTV_Power "Power" { channel="lgwebos:WebOSTV:tv:power" }
Player LGTV_Control "Control" { channel="lgwebos:WebOSTV:tv:mediaPlayer" }
String LGTV_Toast "Toast { channel="lgwebos:WebOSTV:tv:toast" }
---

sitemaps/tv.sitemap
---perl
sitemap multimedia label="TV" {
    Switch item=LGTV_Power
    Default item=LGTV_Control visibility=[LGTV_Power==ON]
    Input item=LGTV_Toast icon="screen" inputHint="text" visibility=[LGTV_Power==ON]
}
---

<!-- Add image here -->
```

Reply:
I've added this section and your screenshots from above, thanks!